### PR TITLE
Jenkins improvements for local testing.

### PR DESCRIPTION
**Pull request by gregschohn**
Setup a docker-compose environment to run jenkins and have a couple build options to build with local (current) changes. I'm also setting up/testing some changes for loading the Jenkinsfile from the repo.

### Testing
manual testing

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkinsdocker/buildFromLocal.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            <categories class="java.util.concurrent.CopyOnWriteArrayList"/>
+            <throttleEnabled>false</throttleEnabled>
+            <throttleOption>project</throttleOption>
+            <limitOneJobWithMatchingParams>false</limitOneJobWithMatchingParams>
+            <paramsToUseForLimit/>
+        </hudson.plugins.throttleconcurrents.ThrottleJobProperty>
+
+    </properties>
+    <definition class="org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition" plugin="workflow-cps@3894.3896.vca_2c931e7935">
+        <script>pipeline {
```
Comment (peternied) [inline jenkinsdocker/buildFromLocal.xml:30]: Can we move this to a separate file for easier reading / formatting?

**Review by peternied — COMMENTED** [review]
Couple of quick comments, when I get a chance I'd like to spin this up too.

File: jenkinsdocker/buildFromLocal.xml (hunk 1 above)
Comment (gregschohn) [reply jenkinsdocker/buildFromLocal.xml]: Not really as far as I can tell... I wish.  I've spent days trying to figure out how to do this w/out requiring a checkin to git first.
That said, I think that we'll just be loading from SCM & requiring the extra commit to test any jenkins changes.  Note that I'm trying to do the local changes clone from the main (SCM checked out) pipeline now

**Review by mikaylathompson — APPROVED** [review]

File: jenkinsdocker/plugins.txt
[the remark this answers is not in the document]
Comment (peternied) [reply jenkinsdocker/plugins.txt:1]: Thanks for doing this - all the difference between what we've already been deploying look sensible

**Comment by gregschohn** [comment]
The conents of this PR were merged in another squashed merge by mikaylathompson.

**Pull request closed without merging**